staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -539,21 +542,25 @@ func (p *processorListener) pop() {
 	}
 }
 
-func (p *processorListener) run() {
-	defer utilruntime.HandleCrash()
-
-	for next := range p.nextCh {
-		switch notification := next.(type) {
-		case updateNotification:
-			p.handler.OnUpdate(notification.oldObj, notification.newObj)
-		case addNotification:
-			p.handler.OnAdd(notification.newObj)
-		case deleteNotification:
-			p.handler.OnDelete(notification.oldObj)
-		default:
-			utilruntime.HandleError(fmt.Errorf("unrecognized notification: %#v", next))
+func (p *processorListener) run(stopCh <-chan struct{}) {
+	// this call blocks until the channel is closed.  When a panic happens during the notification
+	// we will catch it, **the offending item will be skipped!**, and after a short delay (one second)
+	// the next notification will be attempted.  This is usually better than the alternative of never
+	// delivering again.
+	wait.Until(func() {
+		for next := range p.nextCh {
+			switch notification := next.(type) {
+			case updateNotification:
+				p.handler.OnUpdate(notification.oldObj, notification.newObj)
+			case addNotification:
+				p.handler.OnAdd(notification.newObj)
+			case deleteNotification:
+				p.handler.OnDelete(notification.oldObj)
+			default:
+				utilruntime.HandleError(fmt.Errorf("unrecognized notification: %#v", next))
+			}
 		}
```

Comment:
This loop runs until `p.nextCh` is closed which means normal graceful shutdown. So, instead of adding `stopCh` as a parameter to the func, it is possible to make a channel right before `wait.Until()` is invoked and `close()` it right after the `for` loop. Does it make sense?